xarray/core/computation.py

```diff
@@ -1299,3 +1299,49 @@ def where(cond, x, y):
         dataset_join="exact",
         dask="allowed",
     )
+
+
+def polyval(coord, coeffs, degree_dim="degree"):
+    """Evaluate a polynomial at specific values
+
+    Parameters
+    ----------
+    coord : DataArray or array-like
+        The 1D coordinate along which to evaluate the polynomial.
+    coeffs : DataArray or array-like
+        Coefficients of the polynomials.
+    degree_dim : str or int, optional
+        Name of dimension or axis number along which the degree increases in `coeffs`.
+
+    See also
+    --------
+    `xarray.DataArray.polyfit`
+    `numpy.polyval`
+    """
+    from .dataarray import DataArray
+    from .missing import get_clean_interp_index
+
+    if not isinstance(coord, DataArray):
+        coord = DataArray(coord, dims=("x",), name="x")
```

Comment:
We don't use `'x'` as a default elsewhere - if you instead specify that `coord` must be an `xarray.DataArray` then a dimension name won't have to be decided arbitrarily.

Reply:
Makes sense, this simplifies the function considerably.